### PR TITLE
[REG-1570] Brief docs for writing custom discoverers and bots

### PR DESCRIPTION
**Pull request by analogrelay**
This is a brief doc to give people the basic information on how to create custom Entity Discoverers and Bots. It's **not** intended to be a complete reference. There are definitely a few things left to the reader. We can enhance these docs as we go.

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: docs/automated-testing/creating-custom-components.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+* An ID that must be unique across all entity instances in the game
+* A name, usually derived from the GameObject name, that does not need to be unique.
+* A type, which is usually derived from the .NET Type Name of the Component backing the entity.
+* A description, which can be presented to users, **or** to Large Language Models (LLMs) to assist in understanding the purpose of the entity.
+* A set of States representing relevant properties of the entity
+* A set of Actions representing things the bot can _do_ with the entity.
+
+A State is a simple key/value pair that represents some value that will be relevant to a bot.
+For example, a Canvas Group exposes it's `interactable` property as a state value, to allow a bot to identify if the group is interactable.
```
Comment (abeizer) [inline docs/automated-testing/creating-custom-components.mdx:33]:
```suggestion
For example, a Canvas Group exposes its `interactable` property as a state value, to allow a bot to identify if the group is interactable.
```

File: docs/automated-testing/creating-custom-components.mdx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+namespace MyGame
+{
+    public class MyCustomDiscoverer : EntityDiscoverer
+    {
+    }
+}
+```
+
+An Entity Discoverer is a Unity component like any other, so if you want, you can provide handlers for any of Unity's events (`Awake`, `Update`, etc.).
+However, the base class provides a basic pattern we recommend following if you can.
```
Comment (abeizer) [inline docs/automated-testing/creating-custom-components.mdx:60]: I like this inclusion of... here's our "official" pattern for how you should do it, but if that doesn't work for your game then you have access to these other standard Unity concepts. I think we should continue adding tidbits like this going forward!

**Review by abeizer — APPROVED** [review]

File: docs/automated-testing/creating-custom-components.mdx [hunk 3]
```diff
@@ -0,0 +1,192 @@
+---
+sidebar_label: Custom Bots and Discoverers
+---
+
+# Custom Bots and Discoverers
+
+If the built in Bot and Discoverer components aren't sufficient for your testing needs, you can also create your own with just a few steps and some code.
```
Comment (addisonbgross) [inline docs/automated-testing/creating-custom-components.mdx:7]:
```suggestion
If the built-in Bot and Discoverer components aren't sufficient for your testing needs, you can also create your own with just a few steps and some code.
```

**Review by addisonbgross — APPROVED** [review]
Solid docs 👍🏻 
Just a couple of non-blocking comments

**Review by addisonbgross — APPROVED** [review]

**Review by abeizer — APPROVED** [review]
